TESTING.asciidoc

```diff
@@ -586,6 +586,85 @@ repository without fetching latest. For these use cases, you can set the system
 property `tests.bwc.git_fetch_latest` to `false` and the BWC builds will skip
 fetching the latest from the remote.
 
+== How to write good tests?
+
+=== Base classes for test cases
+
+There are multiple base classes for tests:
+ - **`ESTestCase`**: The base class of all tests. It is typically extended
+   directly by unit tests.
+ - **`ESSingleNodeTestCase`**: This test case sets up a cluster that has a
+   single node.
+ - **`ESIntegTestCase`**: An integration test case that creates a cluster that
+   might have multiple nodes.
+ - **`ESRestTestCase`**: An integration tests that interacts with an external
```

Comment:
Small comment: would it be worth clarifying that this includes YAML tests, as that is a very common way in which REST tests are encountered?